src/legacy/core_plugins/metrics/server/lib/vis_data/helpers/unit_to_seconds.js

```diff
@@ -29,27 +30,45 @@ const units = {
   y: 86400 * 7 * 4 * 12, // Leap year?
 };
 
+const sortedUnits = sortBy(Object.keys(units), key => units[key]);
+
 export const parseInterval = (intervalString) => {
   let value;
   let unit;
 
   if (intervalString) {
     const matches = intervalString.match(INTERVAL_STRING_RE);
 
-    value = Number(matches[1]);
-    unit = matches[2];
+    if (matches) {
+      value = Number(matches[1]);
+      unit = matches[2];
+    }
   }
 
   return { value, unit };
 };
 
-export const convertIntervalToUnit = (intervalString, unit) => {
+export const convertIntervalToUnit = (intervalString, newUnit) => {
   const parsedInterval = parseInterval(intervalString);
-  const value = Number((parsedInterval.value * units[parsedInterval.unit] / units[unit]).toFixed(2));
+  let value;
+  let unit;
+
+  if (parsedInterval.value && units[newUnit]) {
+    value = Number((parsedInterval.value * units[parsedInterval.unit] / units[newUnit]).toFixed(2));
+    unit = newUnit;
+  }
 
   return { value, unit };
 };
 
+export const getSuitableUnit = intervalInSeconds => sortedUnits.find((key, index, array) => {
+  const nextUnit = array[index + 1];
+  const isValidInput = isNumber(intervalInSeconds) && intervalInSeconds > 0;
+  const isLastItem = index + 1 === array.length;
+
+  return isValidInput && (intervalInSeconds >= units[key] && intervalInSeconds < units[nextUnit] || isLastItem);
+});
+
 export default (unit) => {
```

Comment:
I propose to replace it by simple object access `UNITS[unit]`.

Reply:
Thanks! Default export was removed